baseplate/secrets/store.py

```diff
@@ -142,6 +161,58 @@ def get_raw(self, path):
         except KeyError:
             raise SecretNotFoundError(path)
 
+    def get_credentials(self, path):
+        """Decode and return a credential secret.
+
+        Credential secrets are a convention of username/password pairs stored as
+        separate values in the raw secret payload.
+
+        The following keys are significant:
+
+        ``type``
+            This must always be ``credential`` for this method.
+        ``encoding``
+            This must be un-set or set to ``identity``.
+        ``username``
+            This contains the raw username.
+        ``password``
+            This contains the raw password.
+
+        :rtype: :py:class:`CredentialSecret`
+
+        """
+        secret_attributes = self.get_raw(path)
+
+        if secret_attributes.get("type") != "credential":
+            raise CorruptSecretError(path, "secret does not have type=credential")
+
+        encoding = secret_attributes.get("encoding", "identity")
+
+        if encoding != "identity":
+            raise CorruptSecretError(path,
+                                    "secret has encoding=%s rather than "
+                                    "encoding=identity" % encoding)
+
+        missing_keys = []
+        values = {}
+        for key in ("username", "password"):
+            try:
+                val = secret_attributes[key]
+                if not isinstance(val, string_types):
+                    raise CorruptSecretError(path, "secret value '%s' is not a string" % key)
+                values[key] = val
+            except KeyError:
+                missing_keys.append(key)
+
+        if missing_keys:
+            if len(missing_keys) > 1:
+                message = "keys %s" % (tuple(missing_keys),)
+            else:
+                message = "key '%s'" % missing_keys[0]
+            raise CorruptSecretError(path, "secret does not have %s" % message)
```

Comment:
Maybe I shouldn't get fancy with this and just `raise` at the first `KeyError`?

Reply:
oh wow, even with proper pluralization. yeah that's probably a bit overkill for an exception message hah. up to you.